server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -207,7 +208,20 @@ public SegmentInfos readLastCommittedSegmentsInfo() throws IOException {
         failIfCorrupted();
         try {
             return readSegmentsInfo(null, directory());
-        } catch (CorruptIndexException | IndexFormatTooOldException | IndexFormatTooNewException ex) {
+        } catch (IndexFormatTooOldException e) {
+            // Temporary workaround
```

Comment:
could you expand on why we need this workaround? I could see how we need special logic for the case where we read from a specific index commit, but given the argument here is null, I would suggest to just change Store#readSegmentsInfo to provide the major argument at all times and then we no longer need this special handling? We do need to handle the case for reading the from a specific index comment pending 10.1 though because the method is not yet public in Lucene

Reply:
Same for all the special casing done in this class, I think that we should centralize the segment infos parsing in a single place, and always provide the min supported major argument, I don't see why not. Not doing so complicates things for no reason?

Reply:
Yeah sorry, I've been a bit nuclear here to have the tests passed.

I reworked this in the latest commit so that all methods finally end using one of the `Lucene#readSegmentInfos()` methods. I implemented a workaround there for reading a specific commit in N-2 version, let me know what you think.